SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -295,6 +297,9 @@ void FLocalDeploymentManager::TryStartLocalDeployment(FString LaunchConfig, FStr
 		// Timeout detection.
 		if (bStartingDeployment && Output.Contains(TEXT("startup completed")))
 		{
+			// IMPORTANT - bSuccessfullyStartedRuntime must be set before bStartingDeployment to avoid a race later checking against
+			// bSuccessfullyStartedRuntime.
+			bSuccessfullyStartedRuntime = true;
```

Comment:
I think this is run on a thread there may still be a race due to ordering of reads/writes not being guaranteed with eventual consistency - atomic read/writes of these values would resolve that (by having acquire/release memory barrier operations) :blobsweat: 

One possible solution without too much effort: Combining these values into a single variable so they are either both fully written or fully not written. Atomics would ensure cache is not stale and compiler optimizations around the read not possible (like reading the value once since because it has no barriers, subsequent reads can be omitted)